src/libos/src/fs/timer_file.rs

```diff
@@ -0,0 +1,183 @@
+use super::*;
+
+use crate::time::{clockid_t, itimerspec_t, timespec_t, ClockID};
+use atomic::{Atomic, Ordering};
+use std::time::Duration;
+
+/// Native Linux timerfd
+#[derive(Debug)]
+pub struct TimerFile {
+    host_fd: HostFd,
+    host_events: Atomic<IoEvents>,
+    notifier: IoNotifier,
+}
+
+impl TimerFile {
+    pub fn new(clockid: ClockID, flags: TimerCreationFlags) -> Result<Self> {
+        let raw_host_fd = try_libc!({
+            let mut ret: i32 = 0;
+            let status = occlum_ocall_timerfd_create(&mut ret, clockid as clockid_t, flags.bits());
+            assert!(status == sgx_status_t::SGX_SUCCESS);
+            ret
+        }) as FileDesc;
+        let host_fd = HostFd::new(raw_host_fd);
+        let host_events = Atomic::new(IoEvents::empty());
+        let notifier = IoNotifier::new();
+        Ok(Self {
+            host_fd,
+            host_events,
+            notifier,
+        })
+    }
+
+    pub fn host_fd(&self) -> c_int {
+        self.host_fd.to_raw() as c_int
+    }
+
+    pub fn set_time(&self, flags: TimerSetFlags, new_value: &itimerspec_t) -> Result<itimerspec_t> {
+        let mut ret = 0;
+        let mut old_value: itimerspec_t = Default::default();
+        let fd = self.host_fd() as FileDesc;
+        let sgx_status = unsafe {
+            occlum_ocall_timerfd_settime(&mut ret, fd, flags.bits(), new_value, &mut old_value)
+        };
+        assert!(sgx_status == sgx_status_t::SGX_SUCCESS);
+        assert!(ret == 0 || libc::errno() == Errno::EINTR as i32);
+        if ret != 0 {
+            return_errno!(EINTR, "settime interrupted");
+        }
+        Ok(old_value)
+    }
+
+    pub fn time(&self) -> Result<itimerspec_t> {
+        let mut ret = 0;
+        let mut curr_value: itimerspec_t = Default::default();
+        let fd = self.host_fd() as FileDesc;
+        let sgx_status = unsafe { occlum_ocall_timerfd_gettime(&mut ret, fd, &mut curr_value) };
+        assert!(sgx_status == sgx_status_t::SGX_SUCCESS);
+        assert!(ret == 0 || libc::errno() == Errno::EINTR as i32);
+        if ret != 0 {
+            return_errno!(EINTR, "gettime interrupted");
+        }
+        Ok(curr_value)
+    }
+}
+
+bitflags! {
+    pub struct TimerCreationFlags: i32 {
+        /// Provides semaphore-like semantics for reads from the new file descriptor
+        /// Non-blocking
+        const TFD_NONBLOCK  = 1 << 11;
+        /// Close on exec
+        const TFD_CLOEXEC   = 1 << 19;
+    }
+}
+
+bitflags! {
+    pub struct TimerSetFlags: i32 {
+        const TFD_TIMER_RELTIME = 0;
```

Comment:
since `TFD_TIMER_RELTIME` is 0, it is useless(it is just the `TimerSetFlags::empty()`), please remove it

Reply:
Done